docs/data-sources/service_principal.md

```diff
@@ -0,0 +1,59 @@
+---
+subcategory: "Security"
+---
+
+# databricks_service_principal Data Source
+
+-> **Note** If you have a fully automated setup with workspaces created by [databricks_mws_workspaces](../resources/mws_workspaces.md) or [azurerm_databricks_workspace](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/databricks_workspace), please make sure to add [depends_on attribute](../index.md#data-resources-and-authentication-is-not-configured-errors) in order to prevent _authentication is not configured for provider_ errors.
+
+Retrieves information about [databricks_service_principal](../resources/service_principal.md).
+
+## Example Usage
+
+Adding service principal `11111111-2222-3333-4444-555666777888` to administrative group
+
+```hcl
+data "databricks_group" "admins" {
+  display_name = "admins"
+}
+
+data "databricks_service_principal" "spn" {
+  application_id = "11111111-2222-3333-4444-555666777888"
+}
+
+resource "databricks_group_member" "my_member_a" {
+  group_id  = data.databricks_group.admins.id
+  member_id = data.databricks_service_principal.spn.id
+}
+```
+
+## Argument Reference
+
+Data source allows you to pick service principals by the following attributes
+
+- `application_id` - (Required) ID of the service principal. The service principal must exist before this resource can be retrieved.
+
+## Attribute Reference
+
+Data source exposes the following attributes:
+
+- `sp_id` - The id of the service principal.
+- `application_id` - This is the application id of the given service principal
+- `external_id` - ID of the service principal in an external identity provider.
```

Comment:
Isn't it the same as application id?

Reply:
they are different - application_id is a uuid but the id is a long

Reply:
I actually meant the external id, hard to review from phone